### PR TITLE
Add a "full lifecycle" FAQ entry to help illustrate the entire process better

### PR DESCRIPTION
**Pull request by tianon**
Hopefully this new section helps make the entire process a bit more clear (and then we can link to it from each of the `README.md` files in the repos under https://github.com/docker-library to help users understand further why images on Docker Hub haven't been updated yet even though the commits appear to exist).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,22 +32,44 @@ This question is so common it's answered in our primary repository!
 
 See [the answer in the readme of the `github.com/docker-library/official-images` repository](https://github.com/docker-library/official-images#what-do-you-mean-by-official).
 
+### An image's source changed in Git, now what?
```
Comment (psftw) [inline README.md]: missing from table of contents above

Comment (tianon) [reply README.md]: We've got a bot that'll do that post-merge 😅

(So if we don't include that part in PRs they're less likely to have conflicts, which is relevant given I have two in-flight ATM. 😇)

File: README.md [hunk 2]
```diff
@@ -32,22 +32,44 @@ This question is so common it's answered in our primary repository!
...
+
+	-	for [`golang`](https://hub.docker.com/_/golang), that would be [the `put-shared/light` job](https://doi-janky.infosiftr.net/job/put-shared/job/light/), because it is not [a "heavy hitter"](https://github.com/docker-library/official-images/blob/master/heavy-hitters.txt)
+
+For images [maintained by the docker-library team](https://github.com/docker-library), we typically include a couple useful scripts in the repository itself (`./update.sh` and `./generate-stackbrew-library.sh`) which help with automating simple version bumps and managing multiple similar versions via `Dockerfile` templating, and [we also have infrastructure which automatically performs those bumps (assuming a build test of the bump passes, which catches most scraping issues, etc)](https://doi-janky.infosiftr.net/job/update.sh/) and commits them directly to the relevant image repository (which is exactly how [the illustrative `golang` a9171b commit](https://github.com/docker-library/golang/commit/a9171b851ba926f2979e8c711d25faa025194def) referenced above was created).
```
Comment (yosifkit) [inline README.md]:
```suggestion
For images [maintained by the docker-library team](https://github.com/docker-library), we typically include a couple useful scripts in the repository itself, like `./update.sh` and `./generate-stackbrew-library.sh`, which help with automating simple version bumps via `Dockerfile` templating, and generating the contents of the `library/xxx` manifest file, respectively. [We also have infrastructure which performs those version bumps along with a build and test](https://doi-janky.infosiftr.net/job/update.sh/) and commits them directly to the relevant image repository (which is exactly how [the illustrative `golang` a9171b commit](https://github.com/docker-library/golang/commit/a9171b851ba926f2979e8c711d25faa025194def) referenced above was created).
```

**Comment by tianon** [comment]
Ok, updated with all three suggestions. :+1:

**Pull request merged**